core/fcache.c

```diff
@@ -3841,46 +3886,12 @@ chain_fragments_for_flush(dcontext_t *dcontext, fcache_unit_t *unit, fragment_t
                 continue;
             }
         }
-        ASSERT(f != NULL);
```

Comment:
Seems best to leave this before the next line which de-refs it

Reply:
Done.